src/scripts/main.js

```diff
@@ -1 +1,81 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const id = setTimeout(
+    () => reject(new Error('First promise was rejected')),
+    3000,
+  );
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(id);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    resolve('Second promise was resolved');
```

Comment:
The `secondPromise` resolves on both 'click' and 'contextmenu' events, which means it will resolve twice if both events occur. This could lead to unexpected behavior if the promise is used elsewhere expecting a single resolution. Consider using a flag to ensure it resolves only once.